pkg/didcomm/common/service/destination.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package service
+
+import (
+	diddoc "github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdri"
+)
+
+// Destination provides the recipientKeys, routingKeys, and serviceEndpoint for an outbound message.
+// Can be populated from an Invitation or DIDDoc.
+type Destination struct {
+	RecipientKeys        []string
+	ServiceEndpoint      string
+	RoutingKeys          []string
+	TransportReturnRoute string
+}
+
+// GetDestination constructs a Destination struct based on the given DID and parameters
+// It resolves the DID using the given VDR, and collects relevant data from the resolved DIDDoc.
+func GetDestination(did, serviceType, keyType string, vdr vdri.Registry) (*Destination, error) {
+	didDoc, err := vdr.Resolve(did)
+	if err != nil {
+		return nil, err
+	}
+
+	return MakeDestination(didDoc, serviceType, keyType)
+}
+
+// MakeDestination makes a Destination object from a DID Doc, following the given parameters.
+func MakeDestination(didDoc *diddoc.Doc, serviceType, keyType string) (*Destination, error) {
+	didCommService, err := diddoc.GetDIDCommService(didDoc, serviceType)
```

Comment:
FYI: I don't need these functions anymore :) I am going to use `SendToDID(msg interface{}, myDID, theirDID string)`

Reply:
I'm using GetDestination inside SendToDID now :)

And MakeDestination is still useful for DIDExchange.

Reply:
just to clarify, `GetDID` function always returns myDID, right?.
e.g `GetDID(myDID) returns myDID` or `GetDID(theirDID) returns myDID`